File: 2021/day1/day1.py


with open("./input.txt") as f:
    lines = f.readlines()

    lines = list(map(int, lines))
    
    depth1 = 0
    depth2 = 0
    win = []


    for x in range(1, len(lines)):
        if lines[x - 1] < lines[x]:
            depth1 += 1

    for x in range(0, len(lines)-2):
        y = lines[x] + lines[x+1] + lines[x+2]
        win.append(y)
    
    for x in range(1, len(win)):
        if win[x - 1] < win[x]:
            depth2 += 1
            

    print(depth1)
    print(depth2)

File: 2021/day1/day1.hs



main :: IO ()
main = do
    lines <- readFile "input.txt"
    let input = map (read::String -> Int) (words lines)

    --part 1
    print $ length (filter (uncurry (<)) (zip input (tail input)))

    --part 2
    let list = map (\(x,y,z) -> x+y+z) (zip3 input (tail input) (tail (tail input)))
    print $ length (filter (uncurry (<)) (zip list (tail list)))

    
    
    
            